src/markdown.py

```diff
@@ -42,13 +42,18 @@ def merge_dict(dict_1, dict_2: Dict[str, str]) -> Dict[str, List[str]]:
         dict_1 = {k: [] for k in dict_2.keys()}
     elif isinstance(list(dict_1.values())[0], list) is False:
         dict_1 = {k: [v] for k, v in dict_1.items()}
+    for key in dict_2.keys():
+        if key not in dict_1:
+            dict_1[key] = []
+            for f in range(0, len(list(dict_1.values())[0])):
+                dict_1[key].append("")
+
     for key in dict_1.keys():
         try:
             dict_1[key].append(dict_2[key])
         except KeyError as err:
             dict_1[key].append("")
             logging.warning(f"Key does not exist: {err}")
```

Comment:
I would like to preserve the original `merge_dict` function so that the path input can still work in the same way. Perhaps make this into another function and clarify all the keys are preserved, or have an argument so that it could be specified whether to preserve all keys. This way, only the file uploader would use the updated function, while the old file import would use the same algorithm as before. Please let me know if this is not clear. Thanks!